Create Project Overview Notebook

BryanCutler opened the pull request:
This adds a notebook to show the basics and give an overview of Text Extensions for Pandas, and serve as a jumping off point to other notebooks.

BryanCutler commented:
I still need to fill in some parts, but could use a review of this so far. cc @frreiss @MLnick @kmh4321

frreiss commented on notebooks/Text_Extensions_for_Pandas_Overview.ipynb:
```diff
@@ -0,0 +1,1543 @@
+{
```
<p>convienent --&gt; convenient</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

frreiss replied:
<p>"begin, end values of the tokens" ==&gt; "begin and end offsets measured in tokens"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

frreiss replied:
<p>Maybe include a reference to https://en.wikipedia.org/wiki/Tensor or to Numpy ndarrays here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

frreiss replied:
<p>"will be incorporated with" =&gt; "supports many"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

frreiss replied:
<p>I like having this section at the end. This is a perfect place to include "teasers" for each of our other notebooks, with links to the notebooks for the reader to find out more.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

frreiss commented on notebooks/Text_Extensions_for_Pandas_Overview.ipynb:
```diff
@@ -0,0 +1,1543 @@
+{
```
<p>It would be nice to have an introductory paragraph here that explains the file formats we support and where to find more information about those formats.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

BryanCutler replied:
I only talked about feather, I think that is all that work now right?

frreiss replied:
I think so, yes.

frreiss commented on notebooks/Text_Extensions_for_Pandas_Overview.ipynb:
```diff
@@ -0,0 +1,1543 @@
+{
```
<p>You might want to fold this part into the <code>Span</code> section  above. Perhaps show a single example of something from the <code>spanner</code>  package, then refer to the API docs (https://text-extensions-for-pandas.readthedocs.io/en/latest/#module-text_extensions_for_pandas.spanner) for more information.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

BryanCutler replied:
<p>I came up with an example, I think it flows pretty good but let me know what you think</p>

BryanCutler replied:
It does flow better to have this after the spans, so I moved it there.

BryanCutler commented:
@frreiss I filled in the rest of this notebook and addressed your feedback, please take another look - thanks! Also cc @kmh4321 , maybe you could take a quick look as someone newer to the project and see if everything is clear?

kmh4321 commented:
Sure, I'll also take a look.

On Fri, Sep 25, 2020, 4:58 PM Bryan Cutler <notifications@github.com> wrote:

> @frreiss <https://github.com/frreiss> I filled in the rest of this
> notebook and addressed your feedback, please take another look - thanks!
> Also cc @kmh4321 <https://github.com/kmh4321> , maybe you could take a
> quick look as someone newer to the project and see if everything is clear?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CODAIT/text-extensions-for-pandas/pull/122#issuecomment-699227732>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH24Y4GDV4655DYOKPWFV33SHUVC5ANCNFSM4RRHCF4Q>
> .
>

frreiss commented:
The current version of this notebook looks good to me. Have you had a chance to review it, @kmh4321 ?

kmh4321 commented on notebooks/Text_Extensions_for_Pandas_Overview.ipynb:
```diff
@@ -0,0 +1,1819 @@
+{
```
<ul><li>Might be good to add a hyperlink to Pandas DataFrames -&gt; https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

kmh4321 replied:
<p>The first sentence is a bit confusing. maybe reword it to "Text Extensions for Pandas provides several Pandas extension arrays to the end user. Much of the functionality of this library is built on top of these extension arrays."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

kmh4321 replied:
<p>Add a small blurb about the crude tokenizer in our case is just splitting the paragraph into individual words. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

kmh4321 replied:
<p>This section seems a bit off theme to the overall NLP focus of this notebook. Maybe a line explaining a potential use of TensorArray might be helpful. Or any other rationale behind adding TensorArray support to TEP. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

kmh4321 commented on notebooks/Text_Extensions_for_Pandas_Overview.ipynb:
```diff
@@ -0,0 +1,1819 @@
+{
```
<ul><li>Can we change the requirements to Python 3.6 now that we have support? </li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CODAIT/text-extensions-for-pandas/pull/122/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

BryanCutler replied:
<p>yeah, we should do that on other nbs too</p>

kmh4321 reviewed (approved):
Overall looks great! My comments above are suggestions but not blockers as such.

BryanCutler commented:
Thanks @frreiss and @kmh4321 , good suggestions! I'll update again tonight and merge when done, we can fine tune more later if needed.
